### PR TITLE
Use date time

### PR DESCRIPTION
**Pull request by aososam**
import data from solution .json file with dates as timestamps or string datetimes

### CODE REVIEW

File: python/darpinstances/instance.py [hunk 1]
```diff
@@ -284,30 +284,34 @@ def read_instance(filepath: Path, travel_time_provider: MatrixTravelTimeProvider
         while (line_string):
             line = line_string.split()
             request_id: int = int(index)
-            request_time: int = int(line[0]) / 1000
+            request_timestamp: int = int(line[0]) / 1000
+            request_datetime = datetime.fromtimestamp(request_timestamp)
```
Comment (F-I-D-O) [inline python/darpinstances/instance.py]: all times should be read either from datetime string or timestamp, depending on what is present in the file

File: python/darpinstances/instance.py [hunk 2]
```diff
@@ -284,30 +284,34 @@ def read_instance(filepath: Path, travel_time_provider: MatrixTravelTimeProvider
...
             end_node = Node(int(line[2]))
             equipment = map_equipment_type(line[4]).value if(len(line) > 4) else 0
             min_travel_time = travel_time_provider.get_travel_time(start_node, end_node)
-            max_pickup_time = request_time + int(instance_config['max_prolongation'])
+            max_pickup_timestamp = request_timestamp + int(instance_config['max_prolongation'])
+            max_pickup_datetime = datetime.fromtimestamp(max_pickup_timestamp)
+            min_drop_off_timestamp = request_timestamp + min_travel_time
+            max_drop_off_timestamp = max_pickup_timestamp + min_travel_time
+            min_drop_off_datetime = datetime.fromtimestamp(min_drop_off_timestamp)
+            max_drop_off_datetime = datetime.fromtimestamp(max_drop_off_timestamp)
```
Comment (F-I-D-O) [inline python/darpinstances/instance.py]: same as above

File: python/darpinstances/instance.py [hunk 3]
```diff
@@ -178,7 +178,7 @@ def load_vehicles(vehicles_path: str) -> List[Vehicle]:
     veh_data = darpinstances.inout.load_csv(vehicles_path, "\t")
     vehicles = []
     for index, veh in enumerate(veh_data):
-        vehicles.append(Vehicle(index, Node(int(veh[0])), int(veh[1]),[]))
+        vehicles.append(Vehicle(index, Node(int(veh[0])), int(veh[1]),[], None, None))
```
Comment (F-I-D-O) [inline python/darpinstances/instance.py]: should rather have defaults in the vehicle constructor than explicitly pass None

File: python/darpinstances/solution_checker.py [hunk 4]
```diff
@@ -135,12 +139,12 @@ def check_plan(plan: VehiclePlan, plan_counter: int, instance: DARPInstance, use
                 travel_time = travel_time_provider.get_travel_time(plan.vehicle.initial_position,
                                                                    action_data.action.node)
 
-        time += travel_time
+        time += timedelta(seconds=int(travel_time))
 
         # arrival time check
-        if action_data.arrival_time != time:
+        if action_data.arrival_time.timestamp() != time.timestamp():
```
Comment (F-I-D-O) [inline python/darpinstances/solution_checker.py]: Why not compare datetimes directly?

File: python/darpinstances/solution_checker.py [hunk 5]
```diff
@@ -135,12 +139,12 @@ def check_plan(plan: VehiclePlan, plan_counter: int, instance: DARPInstance, use
...
             print(f"[{plan_counter}. plan, {action_index + 1}. Action] Arrival time mismatch (expected {time}, "
-                  f"was {action_data.arrival_time}) when handling request {action_data.action.request.index}")
+                  f"was {action_data.arrival_time.timestamp()}) when handling request {action_data.action.request.index}")
```
Comment (F-I-D-O) [inline python/darpinstances/solution_checker.py]: should be just time

File: python/darpinstances/solution_checker.py [hunk 6]
```diff
@@ -190,27 +194,27 @@ def check_plan(plan: VehiclePlan, plan_counter: int, instance: DARPInstance, use
 
         # waiting to min time
         if time < action_data.action.min_time:
-            pause_duration = action_data.action.min_time - time
+            pause_duration = action_data.action.min_time.timestamp() - time.timestamp()
```
Comment (F-I-D-O) [inline python/darpinstances/solution_checker.py]: again, should be just min_time - time

File: python/darpinstances/solution_checker.py [hunk 7]
```diff
@@ -190,27 +194,27 @@ def check_plan(plan: VehiclePlan, plan_counter: int, instance: DARPInstance, use
...
             time = action_data.action.min_time
             if(pause_duration > min_pause_length):
                 driving_start = time
 
-        if (time - driving_start > max_pause_interval):
+        if (max_pause_interval and time.timestamp() - driving_start.timestamp() > max_pause_interval):
```
Comment (F-I-D-O) [inline python/darpinstances/solution_checker.py]: and again

File: python/darpinstances/solution_checker.py [hunk 8]
```diff
@@ -190,27 +194,27 @@ def check_plan(plan: VehiclePlan, plan_counter: int, instance: DARPInstance, use
...
             print("in Request {} driver is active {} min, max is {}.".format(action_data.action.request.index, time-driving_start, max_pause_interval))
             plan_ok = False
 
         max_ride_time = instance.darp_instance_config.max_ride_time
 
         #  max ride time check - dropoff
         if max_ride_time and is_drop_off:
-            ride_time = time - departure_times[request.index]
+            ride_time = time.timestamp() - departure_times[request.index].timestamp()
```
Comment (F-I-D-O) [inline python/darpinstances/solution_checker.py]: and again

File: python/darpinstances/solution_checker.py [hunk 9]
```diff
@@ -231,13 +235,13 @@ def check_plan(plan: VehiclePlan, plan_counter: int, instance: DARPInstance, use
     if previous_action and instance.darp_instance_config.return_to_depot:
         travel_time_to_depot = travel_time_provider.get_travel_time(previous_action.node, plan.vehicle.initial_position)
         cost += travel_time_to_depot
-        time += travel_time_to_depot
+        time += timedelta(seconds=int(travel_time_to_depot))
 
     # max route time check
     max_route_duration = instance.darp_instance_config.max_route_duration
-    if max_route_duration and time - plan.departure_time > max_route_duration:
+    if max_route_duration and time.timestamp() - plan.departure_time.timestamp() > max_route_duration:
```
Comment (F-I-D-O) [inline python/darpinstances/solution_checker.py]: and again

File: python/darpinstances/solution_checker.py [hunk 10]
```diff
@@ -231,13 +235,13 @@ def check_plan(plan: VehiclePlan, plan_counter: int, instance: DARPInstance, use
...
         print("[{}. plan] Total max route duration exceeded: Duration is {} but maximum allowed route duration is {}"
-              .format(plan_counter, time - plan.departure_time, max_route_duration))
+              .format(plan_counter, time.timestamp() - plan.departure_time.timestamp(), max_route_duration))
```
Comment (F-I-D-O) [inline python/darpinstances/solution_checker.py]: and finally here

**Review by F-I-D-O — CHANGES_REQUESTED** [review]

File: python/darpinstances/instance.py (hunk 2 above)
Comment (aososam) [reply python/darpinstances/instance.py]: I understood that max_prolongation is a duration not a time, am I wrong?

Comment (F-I-D-O) [reply python/darpinstances/instance.py]: yes, but here you parse dates, not max prolongation

Comment (aososam) [reply python/darpinstances/instance.py]: the only date loaded from the file is request_timestamp which you mentioned on the previous comment so I am not sure which other value you mean in this second comment

Comment (aososam) [reply python/darpinstances/instance.py]: Ok I get the point, it means just to adjust this section after the first comment change is applied

**Review by F-I-D-O — APPROVED** [review]

**Pull request merged**